challenge-manager/chain-watcher/watcher.go

```diff
@@ -1046,19 +1053,90 @@ func (w *Watcher) processEdgeConfirmation(
 	// Check if we should confirm the assertion by challenge winner.
 	challengeLevel := edge.GetChallengeLevel()
 	if challengeLevel == protocol.NewBlockChallengeLevel() {
-		if confirmAssertionErr := w.chain.ConfirmAssertionByChallengeWinner(ctx, protocol.AssertionHash{Hash: common.Hash(claimId)}, edgeId); confirmAssertionErr != nil {
-			return confirmAssertionErr
-		}
-		srvlog.Info("Assertion confirmed by challenge win", log.Ctx{
-			"challengeParentAssertionHash": containers.Trunc(challengeParentAssertionHash.Bytes()),
-		})
+		return w.confirmAssertionByChallengeWinner(ctx, edge, claimId, challengeParentAssertionHash)
```

Comment:
This will block. Instead, we should probably do a goroutine approach like we did with assertions in #545.

Reply:
Done.